File: src/main/java/no/nav/familie/integrasjoner/journalpost/internal/SafError.java
package no.nav.familie.integrasjoner.journalpost.internal;

public class SafError {

    private String message;
    private String exceptionType;
    private String exception;

    public String getMessage() {
        return message;
    }

    public String getExceptionType() {
        return exceptionType;
    }

    public String getException() {
        return exception;
    }

    @Override
    public String toString() {
        return "SafError{" +
               "message='" + message + '\'' +
               ", exceptionType='" + exceptionType + '\'' +
               ", exception='" + exception + '\'' +
               '}';
    }
}

File: src/main/java/no/nav/familie/integrasjoner/journalpost/internal/SafError.kt
@file:Suppress("ktlint:standard:enum-entry-name-case")

package no.nav.familie.integrasjoner.journalpost.internal

data class SafError(
    val message: String,
    val extensions: SafExtension,
)

data class SafExtension(
    val code: SafErrorCode,
    val classification: String,
)

@Suppress("EnumEntryName")
enum class SafErrorCode {
    forbidden,
    not_found,
    bad_request,
    server_error,
}
